olive/common/ort_inference.py

```diff
@@ -2,17 +2,43 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 # --------------------------------------------------------------------------
+# NOTE: Only onnxruntime and its dependencies can be imported here!!!
```

Comment:
Should we update this note? As the onnxruntime should not be dependency of Olive. Also you are using ort as TYPE_CHECKING

Reply:
It is fine to import ort under type checking since it doesn't make ort a required dependency. We do similar things in other places too https://github.com/microsoft/Olive/blob/683ed65a5c11c614759d186c607bcc26553afae3/olive/passes/onnx/transformer_optimization.py#L20 

This file does not make ort a dependency of olive. it is imported lazy when needed insided the functions here. I can update the note to say "imported in this file" instead of "imported here" to be clear.

Reply:
Update note to be clearer.

Reply:
Yes, I know the typing checking.  As you said `This file does not make ort a dependency of olive`. But the comment shows `onnxruntime can be imported here`. 

Yes, `in this file` is better. 
Thanks!